# patch setup.py to use EB values for building
# wpoely86@gmail.com
diff -ur qml-0.2.10.orig/setup.py qml-0.2.10/setup.py
--- qml-0.2.10.orig/setup.py	2017-06-02 18:14:17.000000000 +0200
+++ qml-0.2.10/setup.py	2017-09-07 13:44:12.648192634 +0200
@@ -1,3 +1,4 @@
+import os
 import sys
 from numpy.distutils.core import Extension, setup
 
@@ -16,10 +17,17 @@
 FORTRAN = "f90"
 
 # GNU (default)
-COMPILER_FLAGS = ["-O3", "-fopenmp", "-m64", "-march=native", "-fPIC", 
-                    "-Wno-maybe-uninitialized", "-Wno-unused-function", "-Wno-cpp"]
-LINKER_FLAGS = ["-lgomp"]
-MATH_LINKER_FLAGS = ["-lblas", "-llapack"]
+COMPILER_FLAGS = os.environ.get("CFLAGS", '').split(' ')
+if not COMPILER_FLAGS:
+    COMPILER_FLAGS = ["-O3", "-fopenmp", "-m64", "-march=native", "-fPIC", "-Wno-maybe-uninitialized", "-Wno-unused-function", "-Wno-cpp"]
+
+LINKER_FLAGS = os.environ.get("LDFLAGS", '').split(' ')
+if not LINKER_FLAGS:
+    LINKER_FLAGS = ["-lgomp"]
+
+MATH_LINKER_FLAGS = os.environ.get("LIBLAPACK", '').split(' ')
+if not MATH_LINKER_FLAGS:
+    MATH_LINKER_FLAGS = ["-lblas", "-llapack"]
 
 # For clang without OpenMP: (i.e. most Apple/mac system)
 if sys.platform == "darwin" and all(["gnu" not in arg for arg in sys.argv]):
